processing/src/main/java/io/druid/segment/data/IndexedInts.java

```diff
@@ -21,17 +21,28 @@
 
 import io.druid.query.monomorphicprocessing.CalledFromHotLoop;
 import io.druid.query.monomorphicprocessing.HotLoopCallee;
-import it.unimi.dsi.fastutil.ints.IntIterable;
 
 import java.io.Closeable;
+import java.util.function.IntConsumer;
 
 /**
  * Get a int an index (array or list lookup abstraction without boxing).
+ *
+ * Doesn't extend {@link Iterable} (or {@link it.unimi.dsi.fastutil.ints.IntIterable} to avoid accidential
+ * for-each iteration with boxing.
  */
-public interface IndexedInts extends IntIterable, Closeable, HotLoopCallee
+public interface IndexedInts extends Closeable, HotLoopCallee
 {
   @CalledFromHotLoop
   int size();
   @CalledFromHotLoop
   int get(int index);
+
+  default void forEach(IntConsumer action)
```

Comment:
Is this needed? This is used only in tests.

Reply:
Yes, I added this for nicer code in tests. I think it's ok